documentation/src/main/asciidoc/topics/protostream.adoc

```diff
@@ -0,0 +1,163 @@
+[[protostream]]
+= ProtoStream
+
+link:https://github.com/infinispan/protostream[ProtoStream] is a serialization library based on link:https://developers.google.com/protocol-buffers[Protocol Buffers].
+The link:https://github.com/infinispan/protostream[ProtoStream] library, is responsible for encoding/decoding Java objects
+to/from the Protobuf format and is used extensively throughout {brandname}.
+
+This chapter provides a brief overview of the Protobuf and ProtoStream concepts, before detailing how ProtoStream can be
+used to serialize your Java objects.
+
+== Concepts
+=== .proto Files
+NOTE: Protocol Buffers is a broad subject, we will not detail it here in great detail, so please consult the
+link:https://developers.google.com/protocol-buffers/docs/overview[Developer Guide] for an in-depth
+explanation of the encoding format and best practices.
+
+Protocol Buffers, Protobuf for short, provide a platform independent encoding format that utilises .proto schema files to
+provide a structured representation of entities that can be easily evolved over time whilst maintaining backwards compatibility.
+
+Protobuf is all about structured data, so the first thing todo is to define the structure of your data. This is
+accomplished by declaring Protobug message types in .proto files, as shown in the example below.
+
+[[book_sample_proto]]
+[source,proto]
+.library.proto
+----
+include::config_examples/library.proto[]
+----
+
+In this example .proto file, we define an entity (message type in Protobuf speak) named _Book_, which is contained in the
+package _book_sample_. The _Book_ entity declares several fields of primitive types and a repeatable field (an
+array basically) named _authors_, which is the _Author_ message type also declared in _library.proto_.
+
+There are a few important notes we need to make about Protobuf messages:
+
+* Nesting of messages is possible, but the resulting structure is strictly a tree, never a graph
+* There is no concept of type inheritance
+* Collections are not supported but arrays can be easily emulated using repeated fields
+
+
+=== Marshallers
+As described in the <<Protobuf Schemas,previous section>>, a fundamental concept of the Protobuf format is the defining
+of messages in the .proto schema to determine how an entity is represented. However, in order for our Java
+applications to utilise the Protobuf format to transmit/store data, it's necessary for our Java objects to be encoded.
+This is handled by the ProtoStream library and it's configured _Marshaller_ implemenations, which convert plain old
```

Comment:
s/it's/its/

Reply:
I think "it's" is correct in this case, as we're saying "it is necessary"

Reply:
s/implemenations/implementations/

Reply:
@ryanemerson I'm referring to "its configured"

Reply:
Gotcha, updated.